Source/Initialization/PlasmaInjector.cpp

```diff
@@ -95,22 +95,42 @@ PlasmaInjector::PlasmaInjector (int ispecies, const std::string& name)
     pp.query("density_min", density_min);
     pp.query("density_max", density_max);
 
+    std::string physical_species_s;
+    bool species_is_specified = pp.query("species_type", physical_species_s);
```

Comment:
I think that I would add at least a warning (if not an error) if the user provides a `species_type` and also a `mass` or a `charge`. In this case, I think that the user should know at the beginning of the simulation that `mass` parameter (or `charge`) has been ignored.

Reply:
Currently, of the user specifies both, `<species>.mass` overwrites the mass from `<species>.species_type`, I feel like it is the most natural way to do. I'll add a warning.